main/modes/mode_diceroller.c

```diff
@@ -0,0 +1,635 @@
+                    char rollOutcome[32];
+                    snprintf(rollOutcome,sizeof(rollStr),"%d",diceRoller->fakeVal);
+                    
+                    drawText(
+                        diceRoller->disp,
+                        &diceRoller->ibm_vga8, c555,
+                        rollOutcome,
+                        xGridOffsets[m] - textWidth(&diceRoller->ibm_vga8,rollOutcome)/2,
+                        yGridOffsets[m]
+                    );
+
+                    drawRegularPolygon(xGridOffsets[m],yGridOffsets[m]+5,
+                        polygonSides[diceRoller->rollIndex],-90 + rotationOffsetDeg,20,c555,0
+                    );
+                }
+
+                //char totalStr[32];
+                //snprintf(totalStr,sizeof(totalStr),"Total: %d",total);
+
+                //drawText(
+                //    diceRoller->disp,
+                //    &diceRoller->ibm_vga8, c555,
+                //    totalStr,
+                //   diceRoller->disp->w/2 - textWidth(&diceRoller->ibm_vga8,totalStr)/2,
+                //    diceRoller->disp->h*7/8
+                //);
+
+                int64_t periodUs = 10000000;
+                diceRoller->timeAngle += elapsedUs%periodUs/(double)periodUs*360;
+
+
+                //Cool Nested Spinny Polygons
+                //drawRegularPolygon(diceRoller->disp->w/2, diceRoller->disp->h/2, 3, 3*diceRoller->timeAngle, 50, c555, 0);
+                //drawRegularPolygon(diceRoller->disp->w/2, diceRoller->disp->h/2, 4, 4*diceRoller->timeAngle, 60, c555, 0);
+                //drawRegularPolygon(diceRoller->disp->w/2, diceRoller->disp->h/2, 5, 5*diceRoller->timeAngle, 70, c555, 0);
+                //drawRegularPolygon(diceRoller->disp->w/2, diceRoller->disp->h/2, 6, 6*diceRoller->timeAngle, 80, c555, 0);
+                if(rollAnimationTimeUs > rollAnimationPeriod)
+                {
+                    diceRoller->state = DR_SHOWROLL;
+                }
+                break;
+        }
+        default:
+        {
+            break;
+        }
+    }
+}
+
+void changeActiveSelection()
+{
+    diceRoller->activeSelection = !(diceRoller->activeSelection);
+}
+
+//Never less than 1 except at mode start, never greater than MAXDICE
+void changeDiceCountRequest(int change)
+{
+    diceRoller->requestCount = (diceRoller->requestCount - 1 + change + MAXDICE) % MAXDICE + 1;
+}
+
+void changeDiceSidesRequest(int change)
+{
+    diceRoller->sideIndex = (diceRoller->sideIndex + COUNTCOUNT + change) % COUNTCOUNT;
+    diceRoller->requestSides = validSides[diceRoller->sideIndex];
+}
+
+void doRoll(int count, int sides, int ind)
+{
+    free(diceRoller->rolls);
+    diceRoller->rolls = (int*)malloc(sizeof(int)*count);
+    for(int m=0; m<count; m++)
+    {
+        diceRoller->rolls[m] = (esp_random() % sides) + 1;
+    }
+    diceRoller->rollSize = count;
+    diceRoller->rollSides = sides;
+    diceRoller->rollIndex = ind;
+}
+
+double cosDeg(double degrees)

```

Comment:
If you want to move away from `double`, and also expensive trig functions, we have trig functions based on LUTs that return a sin value between -1024 and 1024:
```C
int16_t getSin1024(int16_t degree);
int16_t getCos1024(int16_t degree);
int32_t getTan1024(int16_t degree);
```